metpy/calc/thermo.py

```diff
@@ -1444,3 +1444,67 @@ def mixed_layer(p, *args, **kwargs):
         ret.append((-1. / actual_depth.m) * np.trapz(datavar_layer, p_layer) *
                    datavar_layer.units)
     return ret
+
+
+@exporter.export
+@check_units('[length]', '[temperature]')
+def dry_static_energy(heights, temperature):
+    r"""Calculate the dry static energy of parcels.
+
+    This function will calculate the dry static energy following the first two terms of
+    equation 3.72 in [Hobbs2006]_.
+
+    Notes
+    -----
+    .. math::\text{dry static energy} = c_{pd} * T + gz
+
+    * :math:`T` is temperature
+    * :math:`z` is height
+
+    Parameters
+    ----------
+    heights : array-like
+        Atmospheric height profile
```

Comment:
It looks like these will work for scalar values, so I don't think profile is correct to use.